sql: introduce EXPLAIN BUNDLE

RaduBerinde opened the pull request:
This is an initial version of EXPLAIN BUNDLE. Similar to EXPLAIN ANALYZE, it
runs the statement but throws away all resulting rows and instead returns a URL
to a zip bundle (which is served via a special HTTP endpoint); example:

```
root@127.249.95.44:37247/movr> EXPLAIN BUNDLE SELECT * FROM promo_codes;
                                  text
------------------------------------------------------------------------
  Download the bundle from:
    http://127.249.95.44:35445/_admin/v1/stmtbundle/535092672054657025
  or from the Admin UI (Advanced Debug -> Statement Diagnostics).
(1 row)
```

The bundle currently contains the statement and `EXPLAIN`, `EXPLAIN (OPT)`
plans. The plan is to add (in future changes) the `OPT, ENV` data, table stats,
`EXPLAIN ANALYZE` diagram, and traces.

This is not yet integrated with Andrei's query diagnostics work other than using
the same system.table. This will be addressed in future changes; the user will
be able to request a bundle+trace from the UI. Conversely, an EXPLAIN BUNDLE
will also fill out the `trace` field in the system table.

Note that the AdminUI will have an interface to list and retrieve collected
bundles.

Release note (sql change): EXPLAIN BUNDLE can be used to run a query and collect
execution information in a support bundle, which can be downloaded via the
AdminUI.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45735)
<!-- Reviewable:end -->

RaduBerinde commented:
@dhartunian @nathanstilwell can you take a look at the http endpoint I set up in `admin,go` to double check I'm not doing something stupid security-wise?

RaduBerinde committed:
sql: introduce EXPLAIN BUNDLE

This is an initial version of EXPLAIN BUNDLE. Similar to EXPLAIN ANALYZE, it
runs the statement but throws away all resulting rows and instead returns a URL
to a zip bundle (which is served via a special HTTP endpoint); example:

```
root@127.249.95.44:37247/movr> EXPLAIN BUNDLE SELECT * FROM promo_codes;
                                  text
------------------------------------------------------------------------
  Download the bundle from:
    http://127.249.95.44:35445/_admin/v1/stmtbundle/535092672054657025
  or from the Admin UI (Advanced Debug -> Statement Diagnostics).
(1 row)
```

The bundle currently contains the statement and `EXPLAIN`, `EXPLAIN (OPT)`
plans. The plan is to add (in future changes) the `OPT, ENV` data, table stats,
`EXPLAIN ANALYZE` diagram, and traces.

This is not yet integrated with Andrei's query diagnostics work other than using
the same system.table. This will be addressed in future changes; the user will
be able to request a bundle+trace from the UI. Conversely, an EXPLAIN BUNDLE
will also fill out the `trace` field in the system table.

Note that the AdminUI will have an interface to list and retrieve collected
bundles.

Release note (sql change): EXPLAIN BUNDLE can be used to run a query and collect
execution information in a support bundle, which can be downloaded via the
AdminUI.

andreimatei reviewed (commented):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-:-M1gabld6fabwo4K_Ish:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @dhartunian, @jordanlewis, @nathanstilwell, and @RaduBerinde)

___
*[pkg/server/admin.go, line 1528 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gYcWd1WzvwvC-bbbl:-M1gYcWd1WzvwvC-bbbm:b-ce473d) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/server/admin.go#L1528)):*
> ```Go
> }
> 
> func (s *adminServer) writeStatementBundle(ctx context.Context, id int64, w http.ResponseWriter) {
> ```

the "write" in the name seems confusing to me. I understand that we're technically writing something to a writer, but I'd name it more like `getStatementBundle`.
___
*[pkg/server/admin.go, line 1575 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gVxxV5K0O2vIHkI7x:-M1gVxxV5K0O2vIHkI7y:bqap0wu) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/server/admin.go#L1575)):*
> ```Go
> 		fmt.Sprintf("attachment; filename=stmt-bundle-%d.zip", id),
> 	)
> 	//w.Header().Set("Content-Type", req.Header.Get("Content-Type"))
> ```

sup with this?
___
*[pkg/server/admin.go, line 1578 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gW4J2-ojMjW2EwBk9:-M1gW4J2-ojMjW2EwBkA:b-1yngh7) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/server/admin.go#L1578)):*
> ```Go
> 
> 	if _, err := io.Copy(w, &bundle); err != nil {
> 		// Too late to return an error now..
> ```

well, if this fails, it fails because there's no longer anybody to return the error to, right?
___
*[pkg/sql/conn_executor_exec.go, line 352 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gWAVC9njvzqxlfwI2:-M1gWAVC9njvzqxlfwI3:b2nigcz) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L352)):*
> ```Go
> 	case *tree.ExplainBundle:
> 		stmt.Statement.AST = s.Statement
> 		// XXX .SQL?
> ```

xxx
___
*[pkg/sql/conn_executor_exec.go, line 353 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gWEid-gc-mB_XL8hb:-M1gWEid-gc-mB_XL8hc:b-xid59q) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L353)):*
> ```Go
> 		stmt.Statement.AST = s.Statement
> 		// XXX .SQL?
> 		discardRows = true
> ```

put a comment here emphasizing that we don't return the rows for the inner query, but instead return a URL. And put a TODO to capture some rows and include them; I think being able to use this functionality through the admin ui to figure out what some requests are returning would be very cool.
___
*[pkg/sql/conn_executor_exec.go, line 491 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gYDhY07Exj7u0Dsq4:-M1gYDhY07Exj7u0Dsq5:b-9in8hy) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L491)):*
> ```Go
> 	}
> 	if explainBundle {
> 		if err := p.generateExplainBundle(ctx, res); err != nil {
> ```

make a note pls that we're not getting here in case of query execution errors
___
*[pkg/sql/explain_bundle.go, line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gZc7C2FW8wzoMpR1G:-M1gZc7DDJHNAgEQcPgF:bngu1z7) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_bundle.go#L30)):*
> ```Go
> // generateExplainBundle creates the bundle and returns the result for an
> // EXPLAIN BUNDLE statement. The result is text containing a URL for the bundle.
> func (p *planner) generateExplainBundle(ctx context.Context, res RestrictedCommandResult) error {
> ```

I looked briefly to see who integral the `planner` is to the whole thing, and I lost track.
To the greatest extent possible, I'd move away from putting more logic in the planner. I think it's fine for the planner to collect information (particularly planning information), but I'd try to pass that information directly into this function. 
___
*[pkg/sql/explain_bundle.go, line 73 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gX12CEAmMvufZsjeA:-M1gX12DDFrbkp7aaVLD:b-5jjzz6) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_bundle.go#L73)):*
> ```Go
> 
> 	err = db.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
> 		// Insert the bundle into system.statement_bundle_chunks.
> ```

while there's a single chunk, I'd modify the reading code to also deal with a single chunk
___
*[pkg/sql/explain_common.go, line 15 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gaUjF1_rH47WCyoaB:-M1gaUjF1_rH47WCyoaC:b-pftlbh) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_common.go#L15)):*
> ```Go
> import "github.com/cockroachdb/cockroach/pkg/sql/opt/memo"
> 
> func (p *planner) FormatOptPlan(flags memo.ExprFmtFlags) string {
> ```

add a comment pls


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-:-M1hHk0C7eHxXq_HFQsg:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @dhartunian, @jordanlewis, and @nathanstilwell)

___
*[pkg/server/admin.go, line 1528 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gYcWd1WzvwvC-bbbl:-M1hCDOt1BzFAUUmILZE:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/server/admin.go#L1528)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

the "write" in the name seems confusing to me. I understand that we're technically writing something to a writer, but I'd name it more like `getStatementBundle`.
</blockquote></details>

Done.
___
*[pkg/server/admin.go, line 1575 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gVxxV5K0O2vIHkI7x:-M1hCSvu7VnTnwRjfnPz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/server/admin.go#L1575)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

sup with this?
</blockquote></details>

Done.
___
*[pkg/server/admin.go, line 1578 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gW4J2-ojMjW2EwBk9:-M1hCTOiDmndZuevEvrZ:bss4a2b) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/server/admin.go#L1578)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

well, if this fails, it fails because there's no longer anybody to return the error to, right?
</blockquote></details>

Removed.
___
*[pkg/sql/conn_executor_exec.go, line 352 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gWAVC9njvzqxlfwI2:-M1hDCtC0lVpedm3Dmlv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L352)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

xxx
</blockquote></details>

Done.
___
*[pkg/sql/conn_executor_exec.go, line 353 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gWEid-gc-mB_XL8hb:-M1hDCWz9gDjQkm0PHxY:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L353)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

put a comment here emphasizing that we don't return the rows for the inner query, but instead return a URL. And put a TODO to capture some rows and include them; I think being able to use this functionality through the admin ui to figure out what some requests are returning would be very cool.
</blockquote></details>

Done.
___
*[pkg/sql/conn_executor_exec.go, line 491 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gYDhY07Exj7u0Dsq4:-M1hDTXZ3YOkEVJ_reNA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L491)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

make a note pls that we're not getting here in case of query execution errors
</blockquote></details>

Done.
___
*[pkg/sql/explain_bundle.go, line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gZc7C2FW8wzoMpR1G:-M1hFK_83Pta6RPhuuiw:b-48vilg) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_bundle.go#L30)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I looked briefly to see who integral the `planner` is to the whole thing, and I lost track.
To the greatest extent possible, I'd move away from putting more logic in the planner. I think it's fine for the planner to collect information (particularly planning information), but I'd try to pass that information directly into this function. 
</blockquote></details>

Good call, done.
___
*[pkg/sql/explain_bundle.go, line 73 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gX12CEAmMvufZsjeA:-M1hEWLJ76UkPpQgnFCF:b-xkqr7j) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_bundle.go#L73)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

while there's a single chunk, I'd modify the reading code to also deal with a single chunk
</blockquote></details>

Why? We can switch to multiple chunks without worrying about whether there's another node with a version that can't read it.
___
*[pkg/sql/explain_common.go, line 15 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gaUjF1_rH47WCyoaB:-M1hFIlf8Rrfut0PD7An:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_common.go#L15)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

add a comment pls
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-:-M1hJxdREJ2XmGuIGU1_:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @dhartunian, @jordanlewis, @nathanstilwell, and @RaduBerinde)

___
*[pkg/sql/explain_bundle.go, line 73 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gX12CEAmMvufZsjeA:-M1hJvDZ9uiMSm7h2vh-:b-y6q23d) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/explain_bundle.go#L73)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Why? We can switch to multiple chunks without worrying about whether there's another node with a version that can't read it.
</blockquote></details>

I see, ok


<!-- Sent from Reviewable.io -->

jordanlewis commented:
I don't want to bikeshed excessively but I'm concerned the name will give us an n+1 problem. I know this one is meant to be the final level of explain, but can we run it past product / docs / SE to see if the name is ok before we set it in stone? @awoods187

I also am still planning to do a review

nathanstilwell reviewed (approved):
LGTM

jordanlewis reviewed (commented):
What about like EXPLAIN ANALYZE ALL? EXPLAIN ALL? EXPLAIN ANALYZE (ALL)? EXPLAIN ANALYZE (BUNDLE)?

Thing is that ANALYZE is well-known to definitely run the query. Not sure if people will expect something caleld BUNDLE to run the query.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-:-M1ksJSv0uCjD5XPV1az:b-vpjawy)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @dhartunian, @jordanlewis, and @RaduBerinde)

___
*[pkg/sql/conn_executor_exec.go, line 352 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gWAVC9njvzqxlfwI2:-M1ksEUh1JlxDDeZJjqP:b82m3ik) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L352)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Done.
</blockquote></details>

what do you mean trimming?
___
*[pkg/sql/explain_bundle.go, line 75 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1ksbZNBoPnXzZYfcSP:-M1ksbZNBoPnXzZYfcSQ:b-xpdgs3) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/explain_bundle.go#L75)):*
> ```Go
> 	fingerprint := tree.AsStringWithFlags(plan.stmt.AST, tree.FmtHideConstants)
> 	statement := tree.AsString(plan.stmt.AST)
> 	description := "query support bundle XXX more info here"
> ```

XXX ^^
___
*[pkg/sql/explain_bundle_test.go, line 37 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1kshtM2yh_8mD6UnGt:-M1kshtM2yh_8mD6UnGu:b5q4pzk) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/explain_bundle_test.go#L37)):*
> ```Go
> 	defer srv.Stopper().Stop(context.Background())
> 	r := sqlutils.MakeSQLRunner(godb)
> 	rows := r.QueryStr(t, "EXPLAIN BUNDLE SELECT 1")
> ```

Can you replace this with a more sophisticated example that actually touches data? Just might catch more things.
___
*[pkg/sql/plan.go, line 361 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1kssQ49bb54t_18L_M:-M1kssQ49bb54t_18L_N:b-albxbg) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/plan.go#L361)):*
> ```Go
> func (p *planTop) formatOptPlan(flags memo.ExprFmtFlags) string {
> 	if p.mem == nil {
> 		return "<no optimizer plan>"
> ```

nit: I think this won't make sense to users... anything we can say in here to help? Maybe an explanation "no optimizer plan: see the xyz foo for more information", I don't know.


<!-- Sent from Reviewable.io -->

dhartunian commented on pkg/sql/explain_bundle.go:
```diff
@@ -0,0 +1,199 @@
+			ctx, "statement-bundle-info-insert", txn,
+			sqlbase.InternalExecutorSessionDataOverride{User: security.RootUser},
+			`INSERT INTO
+			  system.statement_diagnostics(statement_fingerprint, statement, collected_at, bundle_chunks)
```
@RaduBerinde just to think through how the UI would show this, we would query the `statement_diagnostics` table by fingerprint, correct?

RaduBerinde commented:
Jordan, I chose this because I wanted it to be different than all the others because it operates differently and I wanted it to be easy to use. The ALL variants you listed are pretty cryptic. `EXPLAIN ANALYZE (BUNDLE)` I'd be ok with but it's annoyingly long. I'll let @awoods187 chip in too.

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45735)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @dhartunian, and @jordanlewis)

___
*[pkg/sql/conn_executor_exec.go, line 352 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1gWAVC9njvzqxlfwI2:-M1l6tO61UwAmw9kwzb4:bbo14ap) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e3a718b74f5d717b1a10ae0694868bf14cc765e/pkg/sql/conn_executor_exec.go#L352)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

what do you mean trimming?
</blockquote></details>

improved the comment
___
*[pkg/sql/explain_bundle.go, line 75 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1ksbZNBoPnXzZYfcSP:-M1l7-pt6YoQlnPKGj17:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/explain_bundle.go#L75)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

XXX ^^
</blockquote></details>

Done.
___
*[pkg/sql/explain_bundle.go, line 100 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1c3slvwJrH1gvcUy9Z-r2-100:-M1l76-h9j2mgas30A_2:b7u03hh) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/explain_bundle.go#L100)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

@RaduBerinde just to think through how the UI would show this, we would query the `statement_diagnostics` table by fingerprint, correct?
</blockquote></details>

Correct
___
*[pkg/sql/explain_bundle_test.go, line 37 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1kshtM2yh_8mD6UnGt:-M1l7NuU2z3sqhY4oZIT:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/explain_bundle_test.go#L37)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

Can you replace this with a more sophisticated example that actually touches data? Just might catch more things.
</blockquote></details>

Done.
___
*[pkg/sql/plan.go, line 361 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45735#-M1kssQ49bb54t_18L_M:-M1l9_s53zblkdB2tDTN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/958280de870794d5c91a2d7445ee3194296722e8/pkg/sql/plan.go#L361)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

nit: I think this won't make sense to users... anything we can say in here to help? Maybe an explanation "no optimizer plan: see the xyz foo for more information", I don't know.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

awoods187 commented:
I think we should talk this out on Monday. Do we expect appdevs to use this on their own? Will this only be for AppDevs escalating? Bundle captures the gist of what this does--but not its purpose. I'd love a name that focuses more on the outcome we expect to provide to the developer.

jordanlewis reviewed (approved):
LGTM, we can always rename it after code freeze.

RaduBerinde commented:
Thanks! I'm merging and we'll connect on Monday to settle on a name.
